### ISSUE TITLE
Question about training estimator and ReadoutBottleneck

### ISSUE THREAD

**Issue by colorjam** [attribution-bottleneck/attribution-bottleneck-pytorch#1]
Hi, thank you for your interesting work. 
Do you have examples of training the estimator and ReadoutBottelneck?

**Comment by ls1xt** [comment]
Hi, thanks for you interest and sorry for replying so late. The repo was hosted using an throwaway account and I forgot to check it. I think the estimators were trained using a jupyter notebook not included here. We should include it! I invited you to a private easy-to-use version which we will release soon and which has the code to train the estimators.

**Comment by colorjam** [comment]
Thanks. Would you please explain why the calculation of Kl divergence can be computed as:
 ```
0.5 * (tr(noise_cov) + mu ^ T mu - k  -  log det(noise)
```

**Comment by ls1xt** [comment]
sure, it is a KL-divergence for this form: `KL(N(mu, D) || N(0, 1))` where `D` is a diagonal matirx (see [equation 7 from this paper](https://arxiv.org/pdf/1606.05908.pdf#subsection.2.2)). In the paper, it correspond to the `KL(P(Z|R) || Q(Z))` (see [equation (5)](https://openreview.net/pdf?id=S1xWh1rYwB)). As the KL-Divergence stays the same when both distributions are scaled, we normalize Z such that `Q'(Z) = N(0, 1)` by using  `σ(R)`, `μ(R)`. Doing the normalization `P(Z|R)` to get `P'(Z|R)`, we can compute the KL-divergence with: `KL(P'(Z|R) || N(0, 1))`. Hope this makes sense. If you have any further questions, let me know.

**Comment by ls1xt** [comment]
We now made the esay-to-use version public. [Here is a notebook](https://github.com/BioroboticsLab/IBA/blob/master/notebooks/pytorch_IBA_train_readout.ipynb) showing the training of a readout network.
